internal/impl/openai/image_processor.go

```diff
@@ -0,0 +1,193 @@
+				Optional(),
+			service.NewInterpolatedStringField(oipFieldQuality).
+				Description("The quality of the image that will be generated. hd creates images with finer details and greater consistency across the image. This param is only supported for dall-e-3.").
+				Examples("standard", "hd").
+				Advanced().
+				Optional(),
+			service.NewInterpolatedStringField(oipFieldSize).
+				Description("The size of the generated images. Must be one of 256x256, 512x512, or 1024x1024 for dall-e-2. Must be one of 1024x1024, 1792x1024, or 1024x1792 for dall-e-3 models.").
+				Examples("1024x1024", "512x512", "1792x1024", "1024x1792").
+				Advanced().
+				Optional(),
+			service.NewInterpolatedStringField(oipFieldStyle).
+				Description("The style of the generated images. Must be one of vivid or natural. Vivid causes the model to lean towards generating hyper-real and dramatic images. Natural causes the model to produce more natural, less hyper-real looking images. This param is only supported for dall-e-3.").
+				Examples("vivid", "natural").
+				Advanced().
+				Optional(),
+		)
+}
+
+func makeImageProcessor(conf *service.ParsedConfig, mgr *service.Resources) (service.Processor, error) {
+	b, err := makeBaseProcessor(conf)
+	if err != nil {
+		return nil, err
+	}
+	var i *bloblang.Executor
+	if conf.Contains(oipFieldPrompt) {
+		i, err = conf.FieldBloblang(oipFieldPrompt)
+		if err != nil {
+			return nil, err
+		}
+	}
+	var q *service.InterpolatedString
+	if conf.Contains(oipFieldQuality) {
+		q, err = conf.FieldInterpolatedString(oipFieldQuality)
+		if err != nil {
+			return nil, err
+		}
+	}
+	var style *service.InterpolatedString
+	if conf.Contains(oipFieldStyle) {
+		q, err = conf.FieldInterpolatedString(oipFieldStyle)
+		if err != nil {
+			return nil, err
+		}
+	}
+	var size *service.InterpolatedString
+	if conf.Contains(oipFieldSize) {
+		q, err = conf.FieldInterpolatedString(oipFieldSize)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &moderationProcessor{b, i, q, style, size}, nil
+}
+
+type moderationProcessor struct {
+	*baseProcessor
+
+	input   *bloblang.Executor
+	quality *service.InterpolatedString
+	style   *service.InterpolatedString
+	size    *service.InterpolatedString
+}
+
+func (p *moderationProcessor) Process(ctx context.Context, msg *service.Message) (service.MessageBatch, error) {
+	var body oai.ImageGenerationOptions
+	body.DeploymentName = &p.model
+	format := oai.ImageGenerationResponseFormatBase64
+	body.ResponseFormat = &format
+	if p.input != nil {
+		v, err := msg.BloblangQueryValue(p.input)
+		if err != nil {
+			return nil, err
+		}
+		s := bloblang.ValueToString(v)
+		body.Prompt = &s
+	} else {
+		b, err := msg.AsBytes()
+		if err != nil {
+			return nil, err
+		}
+		s := string(b)
+		body.Prompt = &s
+	}
+	if p.quality != nil {
+		r, err := p.quality.TryString(msg)
+		if err != nil {
+			return nil, err
```

Comment:
Might be worth enriching some of these errors via `fmt.Errorf` so users will have an easier time figuring out what failed.